ZohoCRMCharts/Classes/ZCRMFunnel.swift

```diff
@@ -0,0 +1,1148 @@
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[title]-[scrollView]-[conversionRateView]-|", options: [], metrics:  nil, views: ["title": self.titleView, "scrollView": self.scrollView, "conversionRateView": self.conversionRateView])
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|[conversionRateView]|", options: [], metrics: nil, views: ["conversionRateView": self.conversionRateView])
+		self.activate(constraints: constraints)
+		constraints.removeAll()
+		self.stackView1.layoutIfNeeded()
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|[collectionView(collectionViewWidth)]|", options: [], metrics: ["collectionViewWidth": collectionViewWidth], views: ["collectionView": self.collectionView])
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:|[collectionView(collectionViewHeight)]|", options: [], metrics: ["collectionViewHeight": self.stackView1.frame.height], views: ["collectionView": self.collectionView])
+		self.activate(constraints: constraints, true)
+		
+		self.collectionViewFlowLayout.itemSize = CGSize(width: segmentCellWidth, height: self.stackView1.frame.height / self.rowCount.toCGFloat())
+		self.collectionViewFlowLayout.sectionInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+		self.collectionViewFlowLayout.minimumLineSpacing = 0
+		self.collectionViewFlowLayout.minimumInteritemSpacing = 0
+	}
+	
+	
+	private func renderStandardFunnel() {
+		
+		if self.segments.count > 0 {
+			self.renderSegmentFunnel()
+			self.collectionView.bounces = false
+		} else {
+			
+			self.scrollView.translatesAutoresizingMaskIntoConstraints = false
+			self.scrollView.bounces = false
+			self.addSubview(self.scrollView)
+		}
+		
+		self.standardConversionRateView.translatesAutoresizingMaskIntoConstraints = false
+		self.addSubview(self.standardConversionRateView)
+		
+		self.barChart.translatesAutoresizingMaskIntoConstraints = false
+		self.barChart.color = self.renderOptions.barColor
+		self.barChart.valueFontColor = self.renderOptions.valueFontColor
+		self.barChart.titleFontColor = self.renderOptions.stageFontColor
+		self.barChart.rateFontColor = self.renderOptions.rateFontColor
+		self.scrollView.addSubview(self.barChart)
+	}
+	
+	private func updateStandardUIOptions() {
+		
+		self.barChart.valueFontColor = self.renderOptions.valueFontColor
+		self.barChart.titleFontColor = self.renderOptions.stageFontColor
+		self.barChart.rateFontColor = self.renderOptions.rateFontColor
+		self.barChart.color = self.renderOptions.barColor
+		self.barChart.setNeedsLayout()
+		
+		self.standardConversionRateView.font = self.renderOptions.conversionRateFont
+		self.standardConversionRateView.fontColor = self.renderOptions.conversionRateFontColor
+		self.standardConversionRateView.setUIOptions()
+		
+		for label in self.stackView1.arrangedSubviews {
+			(label as! UILabel).font = self.renderOptions.segmentFont
```

Comment:
use if let.

Reply:
There is no any type of subview in that. So we can surely force type cast it.